Observing 533 error for fc-port creation

While creating FC-ports, 533 error is consistently observed specifically on AGS20, AGS20L, and ALFO device models.

Please find the attached docker logs:
```
[07:23:19.534] DEBUG (1): received event: {"ietf-restconf:notification":{"notifications-1-0:object-creation-notification":{"timestamp":"2025-07-18T07:23:19.0+00:00","counter":13,"object-path":"/core-model-1-4:control-construct/forwarding-domain[uuid='VLAN-FD']/fc[uuid='VLAN-700']/fc-port[local-id='9']"}},"event-time":"2025-07-18T07:23:19Z","node-id":"513250006"}
[07:23:19.550] DEBUG (1): starting notification of 1 subscribers for '/v1/notify-device-object-creations', source-stream is OpenDayLight2 -> HTTP://172.28.127.4:8181/rests/notif/device?notificationType=device
[07:23:19.550] DEBUG (1): sending subscriber notification to: HTTP://172.28.127.18:3008/v1/regard-device-object-creation with content: {"notification-proxy-1-0:object-creation-notification":{"counter":13,"timestamp":"2025-07-18T07:23:19.0+00:00","object-path":"/core-model-1-4:network-control-domain=live/control-construct=513250006/forwarding-domain=VLAN-FD/fc=VLAN-700/fc-port=9"}} - debugId: 'df0c7f04-1f67-4641-9afc-5e16aed4b669'
callback : POST http://172.28.127.18:3033/v1/record-service-request header :{"user":"NotificationProxy","originator":"NotificationProxy","x-correlator":"B2eaad82-CafC-0FD8-77EB-fcFACBFB5345","trace-indicator":"1","customer-journey":"unknown","operation-key":"Operation key not yet provided.","Content-Type":"application/json"}body :{"x-correlator":"e9aE7f8b-7AC8-7CC7-Dc06-f1eae8e55fEF","trace-indicator":"1","user":"NotificationProxy","originator":"NotificationProxy","application-name":"NotificationProxy","release-number":"1.0.2","operation-name":"/v1/notify-device-object-creations","response-code":500,"timestamp":"2025-07-18T07:23:21+00:00"}response code:204
[07:23:21.906] ERROR (1): error during subscriber-notification for /v1/notify-device-object-creations - debugId: 'df0c7f04-1f67-4641-9afc-5e16aed4b669'
    err: {
      "message": "Request failed with status code 533",
      "name": "AxiosError",
      "stack":
          AxiosError: Request failed with status code 533
              at settle (/home/openbackhaul/NotificationProxy/node_modules/axios/dist/node/axios.cjs:1913:12)
              at IncomingMessage.handleStreamEnd (/home/openbackhaul/NotificationProxy/node_modules/axios/dist/node/axios.cjs:2995:11)
              at IncomingMessage.emit (node:events:525:35)
              at endReadableNT (node:internal/streams/readable:1358:12)
              at processTicksAndRejections (node:internal/process/task_queues:83:21)
      "config": {
        "transitional": {
          "silentJSONParsing": true,
          "forcedJSONParsing": true,
          "clarifyTimeoutError": false
        },
        "adapter": [
          "xhr",
          "http"
        ],
        "transformRequest": [
          null
        ],
        "transformResponse": [
          null
        ],
        "timeout": 0,
        "xsrfCookieName": "XSRF-TOKEN",
        "xsrfHeaderName": "X-XSRF-TOKEN",
        "maxContentLength": -1,
        "maxBodyLength": -1,
        "env": {
          "Blob": null
        },
        "headers": {
          "Accept": "application/json, text/plain, */*",
          "Content-Type": "application/json",
          "x-correlator": "e9aE7f8b-7AC8-7CC7-Dc06-f1eae8e55fEF",
          "trace-indicator": "1",
          "user": "NotificationProxy",
          "originator": "NotificationProxy",
          "customer-journey": "unknown",
          "operation-key": "Operation key not yet provided.",
          "User-Agent": "axios/1.5.1",
          "Content-Length": "248",
          "Accept-Encoding": "gzip, compress, deflate, br"
        },
        "method": "post",
        "url": "HTTP://172.28.127.18:3008/v1/regard-device-object-creation",
        "data": "{\"notification-proxy-1-0:object-creation-notification\":{\"counter\":13,\"timestamp\":\"2025-07-18T07:23:19.0+00:00\",\"object-path\":\"/core-model-1-4:network-control-domain=live/control-construct=513250006/forwarding-domain=VLAN-FD/fc=VLAN-700/fc-port=9\"}}"
      },
      "code": "ERR_BAD_RESPONSE",
      "status": 533
    }
```

version MWDI_v2.0.1_impl  

while creating fc-port before we are observing 533 error but in latest version after creating fc-port we are observed 533 error and end up with application restrat.

Docker logs:
````
[29.08.2025 05:22.52.241] [LOG]   message {"notification-proxy-1-0:object-creation-notification":{"counter":6,"timestamp":"2025-08-29T05:22:52.0+00:00","object-path":"/core-model-1-4:network-control-domain=live/control-construct=513250006/forwarding-domain=VLAN-FD/fc=VLAN-200/fc-port=2"}} successfully received from topic: device_change_notifications and partition: 0
[29.08.2025 05:22.52.241] [LOG]   {
  'notification-proxy-1-0:object-creation-notification': {
    counter: 6,
    timestamp: '2025-08-29T05:22:52.0+00:00',
    'object-path': '/core-model-1-4:network-control-domain=live/control-construct=513250006/forwarding-domain=VLAN-FD/fc=VLAN-200/fc-port=2'
  }
} -------- device_change_notifications
[05:22:52.246] INFO (1): Send data to Requestor:http://172.28.127.18:4015/core-model-1-4:network-control-domain=live/control-construct=513250006/forwarding-domain=VLAN-FD/fc=VLAN-200/fc-port=2
[29.08.2025 05:22.54.245] [ERROR] InternalServerError: Bad gateway. The resource/service that is addressed does not exist at the device/application.
    at /home/openbackhaul/microWaveDeviceInventory/service/IndividualServicesService.js:7245:19
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  expose: false,
  statusCode: 533,
  status: 533
}
GET /core-model-1-4:network-control-domain=live/control-construct=513250006/forwarding-domain=VLAN-FD/fc=VLAN-200/fc-port=2 533 1954.672 ms - 127
[05:22:54.243] ERROR (1): Post error
    err: {
      "message": "Request failed with status code 409",
      "stack":
          Error: Request failed with status code 409
              at createError (/home/openbackhaul/microWaveDeviceInventory/node_modules/axios/lib/core/createError.js:16:15)
              at settle (/home/openbackhaul/microWaveDeviceInventory/node_modules/axios/lib/core/settle.js:17:12)
              at IncomingMessage.handleStreamEnd (/home/openbackhaul/microWaveDeviceInventory/node_modules/axios/lib/adapters/http.js:269:11)
              at IncomingMessage.emit (node:events:525:35)
              at endReadableNT (node:internal/streams/readable:1358:12)
              at processTicksAndRejections (node:internal/process/task_queues:83:21)
      "config": {
        "url": "http://172.28.127.19:8181/rests/data/network-topology:network-topology/topology=topology-netconf/node=513250006/yang-ext:mount/core-model-1-4:control-construct/forwarding-domain=VLAN-FD/fc=VLAN-200/fc-port=2",
        "headers": {
          "Accept": "application/json, text/plain, */*",
          "Authorization": "Basic bXdkaV90ZXN0YTptd2RpQDEyMw==",
          "User-Agent": "axios/0.21.4"
        },
        "transformRequest": [
          null
        ],
        "transformResponse": [
          null
        ],
        "timeout": 0,
        "xsrfCookieName": "XSRF-TOKEN",
        "xsrfHeaderName": "X-XSRF-TOKEN",
        "maxContentLength": -1,
        "maxBodyLength": -1,
        "transitional": {
          "silentJSONParsing": true,
          "forcedJSONParsing": true,
          "clarifyTimeoutError": false
        },
        "method": "get"
      }
    }
[05:22:54.244] ERROR (1): POST fail!
[05:22:54.267] ERROR (1): unknown
    err: {
      "type": "InternalServerError",
      "message": "unknown",
      "stack":
          InternalServerError: unknown
              at /home/openbackhaul/microWaveDeviceInventory/service/IndividualServicesService.js:11162:17
              at runMicrotasks (<anonymous>)
              at processTicksAndRejections (node:internal/process/task_queues:96:5)
      "expose": false,
      "statusCode": 533,
      "status": 533
    }
node:internal/process/promises:279
            triggerUncaughtException(err, true /* fromPromise */);
            ^

InternalServerError: unknown
    at /home/openbackhaul/microWaveDeviceInventory/service/IndividualServicesService.js:11162:17
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  expose: false,
  statusCode: 533,
  status: 533
}
[testbox@mwtestbox02 ~]$
[testbox@mwtestbox02 ~]$
[testbox@mwtestbox02 ~]$
```

--- a/server/service/IndividualServicesService.js
+++ b/server/service/IndividualServicesService.js
@@ -1229,6 +1229,7 @@ exports.getCachedControlConstruct = function (url, user, originator, xCorrelator
           let objectKey = Object.keys(finalJson)[0];
           finalJson = finalJson[objectKey];
           if (myFields != undefined) {
+            let ccUuid = finalJson[0]['uuid'];
             myFields = decodeURIComponent(myFields);
             var objList = [];
             var rootObj = { value: "root", children: [] }
@@ -1239,6 +1240,7 @@ exports.getCachedControlConstruct = function (url, user, originator, xCorrelator
             if (isJsonEmpty(finalJson)) {
               throw new createHttpError(470, `Resource not existing. Device informs about addressed resource unknown`);
             }
+            finalJson[0]['uuid'] = ccUuid;
           }
           returnObject[objectKey] = finalJson;
         } else {
@@ -10307,7 +10309,7 @@ exports.notifyObjectDeletions = function (url, body, user, originator, xCorrelat
   });
 }
 
-async function getQualityMeasurementSampleNumber(){
+async function getQualityMeasurementSampleNumber() {
   let profileInstance = await utility.getIntegerProfileForIntegerName("qualityMeasurementSampleNumber");
   let qualityMeasurementSampleNumber = profileInstance[onfAttributes.INTEGER_PROFILE.PAC][onfAttributes.INTEGER_PROFILE.CONFIGURATION][onfAttributes.INTEGER_PROFILE.INTEGER_VALUE];
   return qualityMeasurementSampleNumber;
@@ -10332,8 +10334,7 @@ exports.provideCacheQualityStatistics = function (body, user, originator, xCorre
       let cacheQualityStatistics = await ReadRecords("cache-quality-statistics");
       if (cacheQualityStatistics != undefined) {
         let qualityMeasurementSampleNumber = await getQualityMeasurementSampleNumber();
-        if(cacheQualityStatistics["cache-quality-statistics"].length < qualityMeasurementSampleNumber)
-        {
+        if (cacheQualityStatistics["cache-quality-statistics"].length < qualityMeasurementSampleNumber) {
           throw new createHttpError(530, 'Data invalid. Response data not available, incomplete or corrupted');
         }
         if (body) {
@@ -10506,23 +10507,32 @@ exports.provideDataOfLinks = function (body, user, originator, xCorrelator, trac
 exports.provideDeviceStatusMetadata = function (body, user, originator, xCorrelator, traceIndicator, customerJourney) {
   return new Promise(async function (resolve, reject) {
     try {
-      let mountNameListFromRequestBody = body["mount-name-list"];
       let responseMetaDataList = [];
-      await deviceMetadataCacheUpdate.deviceMetaDataListSync();
-      let metaDataListFromElasticSearch = await deviceMetadataCacheUpdate.getDeviceMetaDataList();
-      for (let i = 0; i < mountNameListFromRequestBody.length; i++) {
-        let mountName = mountNameListFromRequestBody[i];
-        for (let j = 0; j < metaDataListFromElasticSearch.length; j++) {
-          let metaDataMountName = metaDataListFromElasticSearch[j]["mount-name"];
-          if (metaDataMountName == mountName) {
-            responseMetaDataList.push(metaDataListFromElasticSearch[j]);
-            break;
+      let mountNameListFromRequestBody = body["mount-name-list"];
+      if (mountNameListFromRequestBody) {
+        if (mountNameListFromRequestBody.length > 0) {
+          await deviceMetadataCacheUpdate.deviceMetaDataListSync();
+          let metaDataListFromElasticSearch = await deviceMetadataCacheUpdate.getDeviceMetaDataList();
+          for (let i = 0; i < mountNameListFromRequestBody.length; i++) {
+            let mountName = mountNameListFromRequestBody[i];
+            for (let j = 0; j < metaDataListFromElasticSearch.length; j++) {
+              let metaDataMountName = metaDataListFromElasticSearch[j]["mount-name"];
+              if (metaDataMountName == mountName) {
+                responseMetaDataList.push(metaDataListFromElasticSearch[j]);
+                break;
+              }
+            }
           }
+          responseMetaDataList = responseMetaDataList.map(metadata => {
+            let stringifiedData = JSON.stringify(metadata);
+            let convertedData = stringifiedData.replace(/:null/g, ':"null"');
+            return JSON.parse(convertedData);
+          });
         }
       }
       resolve({
-          "device-status-metadata": responseMetaDataList
-        });
+        "device-status-metadata": responseMetaDataList
+      });
     } catch (error) {
       console.log(error);
       reject(error);
@@ -10609,9 +10619,9 @@ exports.provideListOfActualDeviceEquipment = function (url, body, user, originat
 exports.provideListOfCachedDevices = function (user, originator, xCorrelator, traceIndicator, customerJourney) {
   return new Promise(async function (resolve, reject) {
     try {
-      
+
       let result = await utility.ReadIdsFromEs();
-      result = result.filter(d=> d != "DeviceMetaDataList").map(d=> ({"mount-name": d}));
+      result = result.filter(d => d != "DeviceMetaDataList").map(d => ({ "mount-name": d }));
       if (result != undefined) {
         const outputJson = {
           "mount-name-list": result
@@ -11111,8 +11121,8 @@ exports.regardDeviceAlarm = function (body) {
       deviceMetadataCacheUpdate.updateMDForPartialCCUpdate(mountname, timeStamp);
       resolve();
     } catch (error) {
-      //logger.error(error);
-      throw error;
+      logger.error(error);
+      reject(error);
     }
   });
 }

--- a/server/service/individualServices/CyclicProcessService/DeviceMetaDataProcess/SlidingWindow.js
+++ b/server/service/individualServices/CyclicProcessService/DeviceMetaDataProcess/SlidingWindow.js
@@ -47,8 +47,8 @@ class SlidingWindow {
                 // start syncing live CC to ES
                 deviceControlConstructUtility.syncControllerCcToEs(device["mount-name"], responseTimeOut, maximumNumberOfRetries)
                     .then(async (response) => {
+                        let currentTime = new Date().toJSON();
                         if (response) {
-                            let currentTime = new Date().toJSON();
                             device["last-complete-control-construct-update-time-attempt"] = currentTime;
                             device["locked-status"] = false;
                             device["exclude-from-qm"] = false;
@@ -95,7 +95,7 @@ async function getNextDeviceMetaDataLocal() {
     try {
         let device = await deviceMetaDataPriorityList.getNextDeviceMetaData();
         return device;
-    } catch(error) {
+    } catch (error) {
         console.log(error);
         return {};
     }
@@ -132,10 +132,9 @@ exports.startSlidingWindowProcessForCCUpdate = async function () {
         console.log('*******************************************************************************************************');
         if (slidingWindowRunner) {
             await exports.stopSlidingWindowProcessForCCUpdate();
-        } else {
-            await initializeDependentIntegerValues();
-            slidingWindowRunner = new SlidingWindow(getNextDeviceMetaDataLocal);
         }
+        await initializeDependentIntegerValues();
+        slidingWindowRunner = new SlidingWindow(getNextDeviceMetaDataLocal);
     } catch (error) {
         console.log(error);
     }

--- a/server/service/individualServices/cacheUpdateBuilder.js
+++ b/server/service/individualServices/cacheUpdateBuilder.js
@@ -31,9 +31,9 @@ exports.cacheUpdateBuilder = function (url, originalJSON, toInsert, hasFilter) {
       if (Array.isArray(currentJSON[key])) {
         // If the field is an array try to find the elemnt with the corrispondent UUID
         const uuidToFind = decodeURIComponent(value);
-        const indexToChange = currentJSON[key].findIndex(item =>
-          (item.uuid && item.uuid === uuidToFind) ||
-          (item['local-id'] && item['local-id'] === uuidToFind)
+        const indexToChange = currentJSON[key].findIndex(item => {
+          return (item.uuid == uuidToFind || item['local-id'] == uuidToFind)
+        }
         );
         if (indexToChange !== -1) {
           // Substitute only the element into the array with the new JSON
@@ -67,7 +67,7 @@ exports.cacheUpdateBuilder = function (url, originalJSON, toInsert, hasFilter) {
         currentJSON = currentJSON[key];
       }
     } else {
-      // logger.trace(`Field not found: ${key}`);
+      lastKey = lastKey + "." + key;
       break;
     }
     i += 1;
@@ -150,7 +150,7 @@ function assignValueToJson(json, path, newJSON, hasFilters) {
               objJSON[arrayName][index] = newJSON[objectKey];
             }
           }
-        } else { 
+        } else {
           // Otherwise go on parsing the object
           objJSON = objJSON[arrayName][index];
         }
@@ -168,15 +168,27 @@ function assignValueToJson(json, path, newJSON, hasFilters) {
             if (newJSON != null) {
               let objectKey = Object.keys(newJSON)[0];
               // if is latest
-              if (i == pathKeys.length -1 &&
-                objectKey.includes(":") && arrayName.includes("-pac")
-              ) {
+              if (i == pathKeys.length - 1 && objectKey.includes(":") && arrayName.includes("-pac")) {
                 logger.debug("This is PAC configuration");
                 let keyLast = objectKey.split(":");
                 objJSON[keyLast[1]] = newJSON[objectKey];
               } else {
                 logger.debug("This doesn't have pac configuration");
-                objJSON[arrayName] = newJSON[objectKey];
+                if (Array.isArray(objJSON[keyToUse])) {
+                  if (Array.isArray(newJSON[objectKey])) {
+                    newJSON[objectKey].forEach(item => {
+                      objJSON[keyToUse].push(item);
+                    })
+                  } else {
+                    objJSON[keyToUse].push(newJSON[objectKey]);
+                  }
+                } else if (typeof objJSON[keyToUse] == "object") {
+                  objJSON[keyToUse] = [...newJSON[objectKey]];
+                } else if (pathKeys.length == 2) {
+                  objJSON[arrayName] = newJSON[objectKey];
+                } else {
+                  objJSON[keyToUse] = newJSON[objectKey];
+                }
               }
             }
           }

